scipy/signal/filter_design.py

```diff
@@ -2179,6 +2183,14 @@ def iirdesign(wp, ws, gpass, gstop, analog=False, ftype='ellip', output='ba',
     wp = atleast_1d(wp)
     ws = atleast_1d(ws)
 
+    if wp.shape[0] != ws.shape[0]:
+        raise ValueError("Passband and stopband frequencies must"
+                         " have the same dimension.")
+    if wp.shape == 2 and not((ws[0] < wp[0] and wp[1] < ws[1]) or
+            (wp[0] < ws[0] and ws[1] < wp[1])):
+        raise ValueError("Passband must lie strictly inside stopband"
+                         " or vice versa")
+
```

Comment:
positive value check for ws, wp is not needed?

Reply:
AFAII `wp, ws` should be from [0,1] range. I'll add this check to function

Reply:
It would be good to add a few lines with tests like:
```
with pytest.raises(ValueError, match="have the same dimension"):
    iirdesign(...)
with pytest.raises(ValueError, match="can't be negative"):
    iirdesign(...)
with pytest.raises(ValueError, match="strictly inside stopband"):
    iirdesign(...)
```

Reply:
Done!